fuel/converters/ilsvrc2010.py

```diff
@@ -0,0 +1,732 @@
+from fuel.server import send_arrays, recv_arrays
+from fuel.utils.logging import (SubprocessFailure, zmq_log_and_monitor,
+                                configure_zmq_process_logger)
+log = logging.getLogger(__name__)
+
+DEVKIT_ARCHIVE = 'ILSVRC2010_devkit-1.0.tar.gz'
+DEVKIT_META_PATH = 'devkit-1.0/data/meta.mat'
+DEVKIT_VALID_GROUNDTRUTH_PATH = ('devkit-1.0/data/'
+                                 'ILSVRC2010_validation_ground_truth.txt')
+PATCH_IMAGES_TAR = 'patch_images.tar'
+TEST_GROUNDTRUTH = 'ILSVRC2010_test_ground_truth.txt'
+TRAIN_IMAGES_TAR = 'ILSVRC2010_images_train.tar'
+VALID_IMAGES_TAR = 'ILSVRC2010_images_val.tar'
+TEST_IMAGES_TAR = 'ILSVRC2010_images_test.tar'
+IMAGE_TARS = TRAIN_IMAGES_TAR, VALID_IMAGES_TAR, TEST_IMAGES_TAR
+
+
+def make_debug_logging_function(logger, process_type, **additional_kwargs):
+    def _debug(status, **kwargs):
+        pid = os.getpid()
+        message_str = "{process_type}({pid}): {status} ".format(
+            process_type=process_type, pid=pid, status=status)
+        message_str += " ".join("{key}={val}".format(key=key, val=val)
+                                for key, val in kwargs.items())
+        kwargs['process_type'] = process_type
+        kwargs['status'] = status
+        kwargs.update(additional_kwargs)
+        logger.debug(message_str, extra=kwargs)
+    return _debug
+
+
+def ilsvrc2010(directory, save_path, image_dim=256, shuffle_train_set=True,
+               shuffle_seed=(2015, 4, 1), num_workers=4, chunk_size=128):
+    """Converter for the ILSVRC2010 dataset.
+
+    Parameters
+    ----------
+    directory : str
+        Path from which to read raw data files.
+    save_path : str
+        Path to which to save the HDF5 file.
+    image_dim : int, optional
+        The number of rows and columns to which images are normalized
+        (default 256).
+    shuffle_train_set : bool, optional
+        If `True` (default), shuffle the training set within the HDF5 file,
+        so that a sequential read through the training set is shuffled
+        by default.
+    shuffle_seed : int or sequence, optional
+        Seed for a `numpy.random.RandomState` used to shuffle the training
+        set order.
+    num_workers : int, optional
+        The number of worker processes to deploy.
+    chunk_size : int, optional
+        The number of images the workers should send to the sink at a
+        time.
+
+    """
+    debug = make_debug_logging_function(log, 'MAIN')
+
+    # Read what's necessary from the development kit.
+    devkit_path = os.path.join(directory, DEVKIT_ARCHIVE)
+    synsets, cost_matrix, raw_valid_groundtruth = read_devkit(devkit_path)
+
+    # Mapping to take WordNet IDs to our internal 0-999 encoding.
+    wnid_map = dict(zip((s.decode('utf8') for s in synsets['WNID']),
+                        xrange(1000)))
+
+    # Mapping to take ILSVRC2010 (integer) IDs to our internal 0-999 encoding.
+    # label_map = dict(zip(synsets['ILSVRC2010_ID'], xrange(1000)))
```

Comment:
But it looks like you're building that later on (`ilsvrc_id_to_zero_based`, lines 123-124).